Documentation/architecture.rst

```diff
@@ -199,243 +235,792 @@ Example
+endpoint will block all connectivity to ports unless explicitly allowed.
+
+
+Orchestration System Specifics
+==============================
+
+Kubernetes
+----------
+
+Cilium regards each deployed Pod as an endpoint with regards to networking and
+security policy enforcement. Labels associated with pods can be used to define
+the identity of the endpoint.
+
+When two pods communicate via a service construct, then the labels of the
+origin pod apply to determine the identity.
+
+Policy Language
+===============
+
+The security policy can be specified in the following formats:
+
+* The Kubernetes *NetworkPolicy* specification which offers to configure a
+  subset of the full Cilium security fun See `Kubernetes Network Policies
+  <https://kubernetes.io/docs/concepts/services-networking/networkpolicies/>`_
+  for details on how to configure Kubernetes network policies. It is possible
+  to define base rules using the Kubernetes specification and then
+  extend these using additional Cilium specific rules.
+* The Cilium policy language as described below. In addition to the what the
+  Kubernetes NetworkPolicy spec supports, the Cilium language allows to
+  implement Layer 7 filtering, deny rules, and hierarchical rules for
+  delegation and precedence purposes. Cilium also provides egress enforcement
+  for Layer 4 and Layer 7 rules.
+
+The data format used by the Cilium policy language is JSON. Additional formats
+may be supported in the future.
+
+Policy consists of a list of rules:
+
+::
 
-TODO: tgraf to complete this section
+	{
+		"rules": [{ rule1, rule2, rule3 }]
+	}
 
-- label-based filtering
-- inbound + outbound filtering rules
-- L3 / L4 filtering (deny behavior is a drop)
-- HTTP-aware filtering (IPv4-only, deny behavior if a HTTP 403)
-- stateful filtering / conn tracking
-- where is filtering performed?
-- Policy persistence / distribution model
-- Policy hierarchy (root, etc.)
-- reserved keywords (e.g., world)
+.. _arch_rules:
 
+Policy Rules
+------------
+
+Multiple types of policy rules are supported, all types following the following
+simple template:
+
+* **coverage:** A list of labels which the endpoint must carry.
+* **rule:** A type specific rule, the following rule types have been
+  implemented:
+
+  * **Allow/Requires:** Connectivity policy, e.g. allow a pod to talk to
+    another pod
+  * **L4** L4 connectivity policy
+
+**Example:**
+
+The following example describes a rule which applies to all endpoints which
+carry the label `id.service1`. 
+
+::
+
+	[{
+		"coverage": ["role=backend"],
+		"allow": allowData
+	}]
+
+Allow Rules
+-----------
+
+This is the simplest rule type. The rule defines a list of labels which are
+allowed to consume whatever endpoints are covered by the coverage.
+
+If an endpoint transmits to another endpoint and the communication is not
+permitted by at least one *allow* rule, all packets of the connection will be
+silently dropped.
```

Comment:
Maybe cilium monitor could be mentioned here on how to debug drops and policy violations. But could be a future section for debugging/introspection perhaps. Could also have the links to the ascii cinema demos.